Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5630,4 +5648,14 @@ std::optional<LayoutUnit> RenderBox::explicitIntrinsicInnerHeight() const
     return std::optional<LayoutUnit> { height->value() };
 }
 
+RenderBlockFlow* RenderBox::blockFormattingContextRoot() const
+{
+    // This method should probably not be called on the initial containing block
+    ASSERT(!is<HTMLHtmlElement>(element()));
+    const auto container = containingBlock();
+    if (!container)
+        return nullptr;
+    return container->createsBlockFormattingContext() ? dynamicDowncast<RenderBlockFlow>(container) : container->blockFormattingContextRoot(); 
```

Comment:
I try to avoid recursive calls when possible (this is essentially a loop on the ancestor chain until you hit the first BFC root)

Reply:
Changed to iterate over the elements instead of recursing